### ISSUE TITLE
Anomaly "Uncaught exception Failure("List.skipn")."

### ISSUE THREAD

**Issue by afdw** [coq/coq#17071]
#### Description of the problem

The code
```coq
Notation "x :||: l" := (true x l) (at level 51).

Compute
  match true with
  | x :||: l => 0
  end.
```
gives the following error:
```
Anomaly
"Uncaught exception Failure("List.skipn")."
Please report at http://coq.inria.fr/bugs/.
```

#### Coq Version

```
user@notebook:~$ coqtop -v
The Coq Proof Assistant, version 8.16.1
compiled with OCaml 4.14.0
```

**Comment by ejgallego** [comment]
Recent master
```
Anomaly "Uncaught exception Failure("List.skipn")."
Please report at http://coq.inria.fr/bugs/.
Raised at Stdlib.failwith in file "stdlib.ml", line 29, characters 17-33
Called from Constrintern.drop_notations_pattern.in_patargs in file "interp/constrintern.ml", line 1803, characters 26-74
Called from Constrintern.drop_notations_pattern.in_not in file "interp/constrintern.ml", line 1855, characters 38-86
Called from Constrintern.intern_cases_pattern in file "interp/constrintern.ml", line 1911, characters 4-53
Called from CList.map in file "clib/cList.ml", line 431, characters 21-24
Called from Constrintern.internalize.intern_multiple_pattern in file "interp/constrintern.ml", line 2346, characters 19-106
Called from CList.map in file "clib/cList.ml", line 431, characters 21-24
Called from Constrintern.internalize.intern_disjunctive_multiple_pattern in file "interp/constrintern.ml", line 2354, characters 15-59
Called from Constrintern.internalize.intern_eqn in file "interp/constrintern.ml", line 2362, characters 22-71
Called from CList.map in file "clib/cList.ml", line 431, characters 21-24
Called from Constrintern.internalize.intern.(fun) in file "interp/constrintern.ml", line 2246, characters 20-68
Called from Constrintern.interp_open_constr in file "interp/constrintern.ml", line 2587, characters 27-65
Called from Vernacentries.vernac_check_may_eval in file "vernac/vernacentries.ml", line 1967, characters 17-106
Called from Vernacentries.translate_vernac.(fun) in file "vernac/vernacentries.ml", line 2523, characters 8-43
Called from Vernacextend.vtreadproofopt.(fun) in file "vernac/vernacextend.ml", line 170, characters 29-44
Called from Vernacinterp.interp_typed_vernac in file "vernac/vernacinterp.ml", line 20, characters 20-113
Called from Vernacinterp.interp_control.(fun) in file "vernac/vernacinterp.ml", line 205, characters 24-69
Called from Flags.with_modified_ref in file "lib/flags.ml", line 17, characters 14-17
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Vernacinterp.interp_gen.(fun) in file "vernac/vernacinterp.ml", line 255, characters 18-43
Called from Vernacinterp.interp_gen in file "vernac/vernacinterp.ml", line 253, characters 6-279
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 81, characters 4-38
Called from Coq__Interp.coq_interp in file "coq/interp.ml" (inlined), line 35, characters 2-29
Called from Coq__Interp.interp.(fun) in file "coq/interp.ml", line 39, characters 16-34
Called from Coq__Protect.eval in file "coq/protect.ml", line 14, characters 14-17
```

**Comment by SkySkimmer** [comment]
c8c93bfea6e3c75ebce256f44043a34fe8933b5e is the first bad commit

**Comment by herbelin** [comment]
Fixed in #17892. Thanks for the report.
